### ISSUE TITLE
Support for SSL

### ISSUE THREAD

**Issue by sorvis** [lujiajing1126/redis-cli#35]
Would it be possible to add support for SSL?

**Comment by lujiajing1126** [comment]
> Would it be possible to add support for SSL?

I suppose the tls support has been added recently. Does https://github.com/lujiajing1126/redis-cli/issues/32#issuecomment-858332137 this comment help?

**Comment by sorvis** [comment]
@lujiajing1126 

Thank you for the fast response that was exactly what I was looking for! I just missed it since its not in the readme. Created a [PR](https://github.com/lujiajing1126/redis-cli/pull/36) to fix that for future me :-)

After utilizing the --tls switch I'm finding this error about having issues verifying the certificate. Is there anything that can be done to allow it to pass by that certificate and accept it?

"failed - unable to verify the first certificate"

**Comment by lujiajing1126** [comment]
> @lujiajing1126
> 
> Thank you for the fast response that was exactly what I was looking for! I just missed it since its not in the readme. Created a [PR](https://github.com/lujiajing1126/redis-cli/pull/36) to fix that for future me :-)
> 
> After utilizing the --tls switch I'm finding this error about having issues verifying the certificate. Is there anything that can be done to allow it to pass by that certificate and accept it?
> 
> "failed - unable to verify the first certificate"

So I understand you are using a custom TLS certificate?

**Comment by sorvis** [comment]
Yes I believe you are right the custom company certificate is what it is complaining about.

**Comment by lujiajing1126** [comment]
> Yes I believe you are right the custom company certificate is what it is complaining about.

I will try to add customized options within this week. Or would you like to contribute this feature?

**Comment by sorvis** [comment]
I took a quick look, but I don't exactly see the how to implement. I did try setting this environment variable and it doesn't crash but it does seem to hang so I'm not sure if that's the right path.

```
export NODE_TLS_REJECT_UNAUTHORIZED=0 && rdcli -h .......

Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
```

**Comment by lujiajing1126** [comment]
> I took a quick look, but I don't exactly see the how to implement. I did try setting this environment variable and it doesn't crash but it does seem to hang so I'm not sure if that's the right path.
> 
> ```
> export NODE_TLS_REJECT_UNAUTHORIZED=0 && rdcli -h .......
> 
> Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
> ```

Generally speaking, there may be two ways,

1. Add option to `rdcli` to skip TLS cert verification,
2. Add option to `rdcli` to allow custom certification.

For `rdcli`, it just exposes options provided by the underlying redis library.
